src/user_main.c

```diff
@@ -98,6 +102,67 @@ void ICACHE_FLASH_ATTR  gpio16_output_set(uint8 value)
         (READ_PERI_REG(RTC_GPIO_OUT) & (uint32)0xfffffffe) | (uint32)(value & 1)
     );
 }
+void beep(int ms)
+{
+    gpio16_output_set(1);
+    os_delay_us(ms*1000);
```

Comment:
This will delay the ADCread routines for half of the second, as beep is called from enableArc<-updatePwm<-system_task
IMHO it would be better to arm a separate timer like the one already for debugging output. Timers work seperate from tasks. And on timer function just disable gpio.
```
os_timer_disarm(&new_timer);
os_timer_setfn(&new_timer, (os_timer_func_t *)disable_speaker, NULL);
os_timer_arm(&new_timer, 500, 0);   // 0 means non  repeating
```

Reply:
removed